.github/workflows/image-builder.yml

```diff
@@ -157,43 +152,45 @@ jobs:
+        if: ${{ false && steps.check-modified.outputs.was-modified == 'true' }}
         uses: docker/build-push-action@v6
         with:
-          context: ${{ inputs.directory }}/${{ matrix.image }}
+          context: ${{ matrix.image == 'prairielearn' && '.' || format('{0}/{1}', inputs.directory, matrix.image) }}
           push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
           # We will tag using a platform-specific image instead of doing a push-by-digest (https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners).
           # We want to use the default docker builder, as buildx uses `docker-container`, which has additional overhead to load local images (https://docs.docker.com/build/ci/github-actions/named-contexts/#using-with-a-container-builder).
           tags: '${{ inputs.image-prefix }}${{ matrix.image }}:${{ github.sha }}-${{ runner.arch }}'
 
-      - name: Export tarball
-        if: ${{ steps.check-modified.outputs.was-modified == 'true' && contains(fromJson(inputs.upload-artifacts-for), format('{0}-{1}', matrix.image, matrix.os)) }}
-        run: |
-          docker save -o ${{ runner.temp }}/${{ matrix.image }}-${{ matrix.os }}.tar "${{ inputs.image-prefix }}${{ matrix.image }}:${{ github.sha }}-${{ runner.arch }}"
-
       - name: Export metadata
-        if: ${{ steps.check-modified.outputs.was-modified == 'true' }}
+        if: ${{ false && steps.check-modified.outputs.was-modified == 'true' }}
         run: |
           mkdir -p ${{ runner.temp }}/image-metadata
           touch ${{ runner.temp }}/image-metadata/${{ matrix.image }}
 
       - name: Upload metadata
-        if: ${{ steps.check-modified.outputs.was-modified == 'true' }}
+        if: ${{ false && steps.check-modified.outputs.was-modified == 'true' }}
         uses: actions/upload-artifact@v4
         with:
           name: image-metadata-${{ inputs.directory}}-${{ matrix.image }}-${{ matrix.os }}
           path: ${{ runner.temp }}/image-metadata/
           if-no-files-found: error
           retention-days: 1
 
-      - name: Upload tarball
-        if: ${{ steps.check-modified.outputs.was-modified == 'true'  && contains(fromJson(inputs.upload-artifacts-for), format('{0}-{1}', matrix.image, matrix.os)) }}
-        uses: actions/upload-artifact@v4
+      # We want to retag the image as `latest` -- so that we can run any potential test suites.
+      - name: Retag image
+        if: ${{ false && steps.check-modified.outputs.was-modified == 'true' }}
+        run: |
+          docker tag "${{ inputs.image-prefix }}${{ matrix.image }}" "${{ inputs.image-prefix }}${{ matrix.image }}:latest"
+
+      # We need to run test suites in the context of the build image, so we define them as composite actions.
+      - name: Test PrairieLearn image
+        if: ${{ format('{0}-{1}', matrix.image, runner.arch) == 'prairielearn-X64' }}
+        uses: './.github/actions/test-prairielearn'
         with:
```

Comment:
Composite actions don't support environment or secret variables, need to be passed in like so.